apps/test/util/wrappedEventListener.js

```diff
@@ -0,0 +1,36 @@
+var original = null;
+var listeners = [];
+
+/**
+ * Wraps window.addEventListener to catch all added listeners, allowing us to
+ * dispose of them when our test finishes
+ */
+module.exports = {
+  /**
+   * Attach our wrapper, tracking all added listeners
+   */
+  attach: function () {
+    if (original) {
+      throw new Error('addEventListener already wrapped');
+    }
+
+    original = window.addEventListener;
+
+    window.addEventListener = function () {
+      listeners.push(arguments);
+      original.apply(window, arguments);
+    };
+  },
+
+  /**
+   * Detach our wrapper, removing all added listeners
+   */
+  detach: function () {
+    window.addEventListener = original;
+    original = null;
+    listeners.forEach(function (argList) {
+      window.removeEventListener.apply(window, argList);
+    });
+    listeners = [];
+  }
+};
```

Comment:
This is a good approach to quickly clean up global event listeners left over by our apps code.
In the future, might it be possible to use this as an observer to enforce the app code to clean up its listeners appropriately and throw a test failure if it doesn't, rather than silently cleaning up after the app? I don't know how feasible it would be to get our current codebase passing, but I was just thinking it might be possible to use this wrapper as a way to remove any event-listener leaks within the apps code, rather than just using it to clean up after those leaks within the test framework.

That would be something nice to have on the way to making our Code Studio modules reloadable/swappable without requiring a full page refresh, for example. (Steve was looking into that possibility a bit, but I don't think he got very far)